### PR TITLE
FIX Jenkins-35847

### PR DESCRIPTION
**Pull request by scherler**
# Description

See [JENKINS-35847](https://issues.jenkins-ci.org/browse/JENKINS-35847).

- based on https://github.com/jenkinsci/blueocean-acceptance-test/pull/26

Refactor gitCommit to return a promise. Add tests for notSet, condensed and notCondensed for a.authors

@jenkinsci/code-reviewers @reviewbybees

### CODE REVIEW

**Comment by michaelneale** [comment]
Looks good to me - this is new tests or also a refactor?

**Comment by scherler** [comment]
there are three new test cases in folder:

- 'Check whether the changes tab shows changes - condensed'
- 'Check whether the changes tab shows changes - one commit':
- blueRunDetailPage.authorsIsNotSet(browser);

but I refactored the gitCommit from the earlier PR to return a Promise and further reduced the problem of race conditions by using callback in async methods. 

I am keen to refactor the whole AT to work with promise the whole way through, so we can chains a test case and be sure it works async. ala:

```
'Validate correct encoding, pipeline graph and steps' : function (browser) {
    // /JENKINS-36616 - Unable to load multibranch projects in a folder
    // JENKINS-36773 / JENKINS-37605 verify encoding and spacing of details
    // JENKINS-36613 Unable to load steps for multibranch pipelines with / in them
    // There should be no authors
    return browser.page.bluePipelineRunDetail().forRun(projectName, 'jenkins', 'feature%2F1', 1)
        .assertTitle('jenkins / ' + anotherFolders.join(' / ') + ' / feature/1')
        .validateGraph()
        .validateSteps(browser)
        .authorsIsNotSet(browser);
}
```

Basically that is working ATM most of the time but there are some code snippet that are not. The above is implicit using then.

Next as well is to look into running our tests in parallel since we starting to have quite a few.
